### PR TITLE
[@mantine/nprogress] Init

### PR DESCRIPTION
**Pull request by jerebtw**
This is a new version of my [old PR](https://github.com/mantinedev/mantine/pull/769) (Complete rewrite)

- [ ] Better prop names
- [ ] Tests
- [ ] Demos
- [ ] Docs

@rtivital can you please check this first version

### CODE REVIEW

File: src/mantine-nprogress/src/NProgressProvider.tsx [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+  transitionDuration = 600,
+  unmountTime = intervalTime + 200,
+  unmountDuration: exitDuration = 600,
+  unmountTransition = 'fade',
+  onFinish,
+}: NProgressProviderProps) {
+  const [_progress, setProgress] = useState(progress);
+  const [mounted] = useDebouncedValue(_progress !== 100, unmountTime);
+
+  const interval = useInterval(
```
Comment (rtivital) [inline src/mantine-nprogress/src/NProgressProvider.tsx:47]: I do not see much purpose in interval, it should be removed.

Comment (jerebtw) [reply src/mantine-nprogress/src/NProgressProvider.tsx:47]: The interval is for the start and stop function so I can increment the progress bar

File: src/mantine-core/src/components/Transition/Transition.tsx [hunk 2]
```diff
@@ -40,6 +43,7 @@ export function Transition({
   duration = 250,
   exitDuration = duration,
   mounted,
+  stayMounted = false,
```
[the remark this answers is not in the document]
Comment (jerebtw) [reply src/mantine-core/src/components/Transition/Transition.tsx:46]: Do you know a way how I can add a transition without unmounting the component

**Review by rtivital — COMMENTED** [review]
All good, please complete remaining tasks

**Comment by jerebtw** [comment]
Close for https://github.com/mantinedev/mantine/pull/1222

**Pull request closed without merging**